### PR TITLE
[Merged by Bors] - feat: port LinearAlgebra.Basis

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Vierkantor** [comment]
Let me take a look for a few hours...

File: Mathlib/LinearAlgebra/Basis.lean [hunk 1]
```diff
@@ -75,6 +75,9 @@ open Function Set Submodule
 
 open BigOperators
 
+-- Porting note: TODO Erase this line.
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Basis.lean:78]:
```suggestion
-- Porting note: TODO Erase this line. (Cf lean4#2074)
```

File: Mathlib/LinearAlgebra/Basis.lean [hunk 2]
```diff
@@ -1218,65 +1239,67 @@ theorem group_smul_span_eq_top {G : Type _} [Group G] [DistribMulAction G R] [Di
...
       simp [Basis.unitsSmul, ← mul_smul]
     simp only [Basis.coord_apply, LinearMap.smul_apply, Basis.repr_self, Units.smul_def,
       SMulHomClass.map_smul, Finsupp.single_apply]
-    split_ifs with h h
+    split_ifs with h
     · simp [h]
     · simp
 #align basis.coord_units_smul Basis.coord_unitsSmul
 
+attribute [-instance] Ring.toNonAssocRing in -- Porting note: TODO: workaround for #2074
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Basis.lean:1293]:
```suggestion
-- Porting note: TODO: workaround for lean4#2074
attribute [-instance] Ring.toNonAssocRing in
```

File: Mathlib/LinearAlgebra/Basis.lean [hunk 3]
```diff
@@ -1528,117 +1564,117 @@ theorem nonzero_span_atom (v : V) (hv : v ≠ 0) : IsAtom (span K {v} : Submodul
...
-    refine' ⟨submodule_eq_Sup_le_nonzero_spans W, _⟩
-    rintro _ ⟨w, ⟨_, ⟨hw, rfl⟩⟩⟩; exact nonzero_span_atom w hw
+    { intro W
+      use { T : Submodule K V | ∃ (v : V) (_ : v ∈ W) (_ : v ≠ 0), T = span K {v} }
+      refine' ⟨submodule_eq_supₛ_le_nonzero_spans W, _⟩
+      rintro _ ⟨w, ⟨_, ⟨hw, rfl⟩⟩⟩; exact nonzero_span_atom w hw}
 
 end AtomsOfSubmoduleLattice
 
+attribute [-instance] Ring.toNonAssocRing -- Porting note: TODO: workaround for #2074
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Basis.lean:1601]:
```suggestion
-- Porting note: TODO: workaround for lean4#2074
attribute [-instance] Ring.toNonAssocRing
```

**Pull request closed without merging**